pkg/strategy/pivotshort/strategy.go

```diff
@@ -197,32 +203,33 @@ func (s *Strategy) Run(ctx context.Context, orderExecutor bbgo.OrderExecutor, se
 	var lastLow fixedpoint.Value
 	futuresMode := s.session.Futures || s.session.IsolatedFutures
 	d := s.CatBounceRatio.Div(s.NumLayers)
-	q := s.Quantity.Div(s.NumLayers)
-	log.Info(futuresMode)
+	q := s.Quantity
+	if !s.TotalQuantity.IsZero() {
+		q = s.TotalQuantity.Div(s.NumLayers)
+	}
+
+	var pivotBuffer []fixedpoint.Value
 
 	session.MarketDataStream.OnKLineClosed(func(kline types.KLine) {
 		if kline.Symbol != s.Symbol || kline.Interval != s.Interval {
 			return
 		}
-		if err := s.activeMakerOrders.GracefulCancel(ctx, s.session.Exchange); err != nil {
-			log.WithError(err).Errorf("graceful cancel order error")
-		}
+
 		if s.pivot.LastLow() > 0. {
 			log.Info(s.pivot.LastLow(), kline.EndTime)
 			lastLow = fixedpoint.NewFromFloat(s.pivot.LastLow())
 		} else {
-			if !lastLow.IsZero() && !s.Position.GetBase().IsZero() {
+			if !lastLow.IsZero() && s.Position.IsShort() && !s.Position.IsDust(kline.Close) {
```

Comment:
can we pull out these blocks to methods can describe their behavior as the method name?

Reply:
on my way